### DISCUSSION TITLE
Getting error " Cannot read properties of undefined (reading '0') "

### DISCUSSION THREAD

**Discussion by umerjunaid** [WalletConnect/walletconnect-monorepo#4328] [Developer Support]
I have used the default code given in docs

`
import { createWeb3Modal } from "@web3modal/wagmi/react";
import { defaultWagmiConfig } from "@web3modal/wagmi/react/config";

import { WagmiProvider } from "wagmi";
import { arbitrum, mainnet, polygon, xdc } from "wagmi/chains";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";

// 0. Setup queryClient
const queryClient = new QueryClient();

// 1. Get projectId
const projectId = process.env.REACT_APP_PROJECT_ID;
if (!projectId) {
  throw new Error("Please set the REACT_APP_PROJECT_ID environment variable");
}
// 2. Create wagmiConfig
const metadata = {
  name: "Web3Modal",
};

const chains = [mainnet, arbitrum, polygon, xdc];
const config = defaultWagmiConfig({
  chains,
  projectId,
  metadata,
});

// 3. Create modal
createWeb3Modal({
  wagmiConfig: config,
  projectId,
  enableAnalytics: true, // Optional - defaults to your Cloud configuration
  enableOnramp: true, // Optional - false as default
});

function App() {
  return (
    <WagmiProvider config={config}>
    // rest of my code 
     </WagmiProvider>
  );
}

export default App;
`

but when compiling it gives error

Cannot read properties of undefined (reading '0')
TypeError: Cannot read properties of undefined (reading '0')
    at defaultWagmiConfig (http://localhost:3000/static/js/bundle.js:124594:34)
    at ./src/App.js (http://localhost:3000/static/js/bundle.js:78:98)
    at options.factory (http://localhost:3000/static/js/bundle.js:149161:31)
    at __webpack_require__ (http://localhost:3000/static/js/bundle.js:148555:33)
    at fn (http://localhost:3000/static/js/bundle.js:148818:21)
    at ./src/index.js (http://localhost:3000/static/js/bundle.js:5257:62)
    at options.factory (http://localhost:3000/static/js/bundle.js:149161:31)
    at __webpack_require__ (http://localhost:3000/static/js/bundle.js:148555:33)
    at http://localhost:3000/static/js/bundle.js:149807:37
    at http://localhost:3000/static/js/bundle.js:149809:12
    
    
    if some one can help me I am new with this 
    
    
![Screenshot from 2024-03-14 01-56-41](https://github.com/WalletConnect/walletconnect-monorepo/assets/64754978/c11ef560-5792-45f8-9dee-e66426199e8e)

**Comment by glitch-txs** [comment]
You'll need to update webpack or use vite. Create react-app has been deprecated.

**Reply by umerjunaid** [reply]
Using wagmi as first time 
As template is in CRA is there any alternative which can be used with CRA?

**Reply by glitch-txs** [reply]
Yes, use Vite: https://vitejs.dev/guide/#scaffolding-your-first-vite-project
